src/core/menu.rs

```diff
@@ -65,15 +65,34 @@ impl App {
 
             if ui.menu_item("Run") {
                 let py_code = self.generate_code();
-                let python_out = Command::new("python3")
+                let mut python_process = match Command::new("python3")
                     .arg("-c")
-                    .arg(py_code)
+                    .arg(&py_code)
                     .arg("--csv")
                     .stdout(Stdio::piped())
                     .spawn()
-                    .unwrap();
+                {
+                    Ok(process) => process,
+                    Err(e) => {
+                        eprintln!("Error: Failed to start python process: {}", e);
+                        return;
+                    }
+                };
+
+                let status = python_process.wait().unwrap();
 
-                self.simulation_state = Some(SimulationState::from_csv(python_out.stdout.unwrap()));
+                if status.success() {
+                    if let Some(output) = python_process.stdout {
+                        self.simulation_state = Some(SimulationState::from_csv(output));
+                    } else {
+                        eprintln!("Error: python process output is not available.");
+                    }
+                } else {
+                    eprintln!(
+                        "Error: python process failed with exit code {}",
+                        status.code().unwrap()
+                    );
+                }
```

Comment:
Gostei das checagens, mas acho que precisaríamos das mesmas coisas ali na plotagem do PDF, que faz umas operações parecidas. Que tal levarmos a execução de Python para uma função separada e chamar ela nesses dois lugares? A função poderia receber uma struct do tipo `Command` diretamente e retornar um resultado com o sucesso contendo a saída do programa e o fracasso contendo o tipo do erro (failed to start/process output unavailable/process failed with exit code)

Reply:
Sugestões feitas no commit c2a14abf89b9699ccb50a2a193ff485b23357ce2
Tive que mudar a função from_csv para receber uma variável tipo String, pra tudo continuar compatível.